python_modules/dagster/dagster_tests/declarative_automation_tests/scenario_utils/asset_daemon_scenario.py

```diff
@@ -350,33 +383,115 @@ def _assert_requested_runs_daemon(self, expected_run_requests: Sequence[RunReque
+    ) -> "AssetDaemonScenarioState":
+        """Asserts that the set of runs requested by the previously-evaluated tick is identical to
+        the set of runs specified in the expected_run_requests argument.
+        """
+        sorted_expected_run_requests = self._assert_run_requets_lists_equal(
+            self.run_requests, expected_run_requests
+        )
+
         if self.is_daemon:
             self._assert_requested_runs_daemon(sorted_expected_run_requests)
 
         return self
 
+    def assert_requested_runs_for_stopped_iteration(
+        self, num_expected_submissions: int, *expected_run_requests: RunRequest
+    ) -> "AssetDaemonScenarioState":
+        """For ticks that are manually stopped during submitting runs, confirm that the number of runs actually submitted
+        before the tick was interrupted is as expected. Also confirm that the requests the tick would have submitted if
+        it were not stopped is as expected.
+        """
+        if not self.is_daemon:
+            assert False, "Cannot stop while submitting runs in non-daemon mode"
+
+        assert len(self.run_requests) == num_expected_submissions
+        # assert that the iteration was actually stopped before all runs were submitted
+        assert len(self.run_requests) < len(expected_run_requests)
+
+        sensor_origin = self.get_sensor_origin()
+        if sensor_origin:
+            origin_id = sensor_origin.get_id()
+            selector_id = sensor_origin.get_selector().get_id()
+        else:
+            origin_id = _PRE_SENSOR_AUTO_MATERIALIZE_ORIGIN_ID
+            selector_id = _PRE_SENSOR_AUTO_MATERIALIZE_SELECTOR_ID
+
+        ticks = sorted(
+            self.instance.get_ticks(
+                origin_id=origin_id,
+                selector_id=selector_id,
+            ),
+            key=lambda tick: tick.tick_id,
+        )
+        latest_tick = ticks[-1]
+        assert len(latest_tick.run_ids) == num_expected_submissions
+        assert (
+            len(latest_tick.unsubmitted_run_ids_with_requests)
+            == len(expected_run_requests) - num_expected_submissions
+        )
+        # since the tick was stopped before all runs were submitted, the tick should be in skipped state
+        assert latest_tick.status == TickStatus.SKIPPED
+
+        if latest_tick.tick_data.reserved_run_ids:
+            assert len(latest_tick.tick_data.reserved_run_ids) == len(expected_run_requests)
+
+        if latest_tick.tick_data.run_requests:
+            self._assert_run_requets_lists_equal(
+                latest_tick.tick_data.run_requests, expected_run_requests
+            )
+        else:
+            if len(expected_run_requests) > 0:
+                self._log_assertion_error(expected_run_requests, [])
+                raise AssertionError("Expected run requests not found in tick data")
+
+        if len(ticks) > 1:
+            prev_tick = ticks[-2]
+            if prev_tick.cursor and latest_tick.cursor:
```

Comment:
TODO - need to add equivalent asserts for non-sensor cursors, but need to figure out a different way to do it because they get overwritten in the db